app/src/main/java/de/danoeh/antennapod/dialog/SubscriptionsFilterDialog.java

```diff
@@ -62,8 +65,59 @@ public static void showDialog(Context context) {
             }
         }
 
-        builder.setPositiveButton(R.string.confirm_label, (dialog, which) -> {
-            filterValues.clear();
+        builder.setPositiveButton(R.string.confirm_label, null);
+        builder.setNeutralButton(R.string.reset, null);
+        builder.setNegativeButton(R.string.cancel_label, null);
+        AlertDialog dialog = builder.create();
+
+        for (int i = 0; i < rows.getChildCount(); i++) {
+            FilterDialogRowBinding binding = FilterDialogRowBinding.bind(rows.getChildAt(i));
+            setFilterButtonListener(binding.filterButton1, rows, dialog);
+            setFilterButtonListener(binding.filterButton2, rows, dialog);
+        }
+
+        dialog.setOnShowListener(dialogInterface -> {
+            Button positiveButton = dialog.getButton(AlertDialog.BUTTON_POSITIVE);
+            positiveButton.setOnClickListener(view -> {
+                filterValues.clear();
+                for (int i = 0; i < rows.getChildCount(); i++) {
+                    if (!(rows.getChildAt(i) instanceof MaterialButtonToggleGroup)) {
+                        continue;
+                    }
+                    MaterialButtonToggleGroup group = (MaterialButtonToggleGroup) rows.getChildAt(i);
+                    if (group.getCheckedButtonId() == View.NO_ID) {
+                        continue;
+                    }
+                    String tag = (String) group.findViewById(group.getCheckedButtonId()).getTag();
+                    if (tag == null) { // Clear buttons use no tag
+                        continue;
+                    }
+                    filterValues.add(tag);
+                }
+                updateFilter(filterValues);
+                dialog.dismiss();
+            });
+
+            Button neutralButton = dialog.getButton(AlertDialog.BUTTON_NEUTRAL);
+            neutralButton.setOnClickListener(view -> {
+                updateFilter(Collections.emptySet());
+                showDialog(context);
+                dialog.dismiss();
+            });
+        });
+        dialog.show();
+        Button neutralButton = dialog.getButton(AlertDialog.BUTTON_NEUTRAL);
+        neutralButton.setEnabled(subscriptionsFilter.isEnabled());
+    }
+
+    private static void updateFilter(Set<String> filterValues) {
+        SubscriptionsFilter subscriptionsFilter = new SubscriptionsFilter(filterValues.toArray(new String[0]));
+        UserPreferences.setSubscriptionsFilter(subscriptionsFilter);
+        EventBus.getDefault().post(new UnreadItemsUpdateEvent());
+    }
```

Comment:
Please move down to reduce the diff, making the PR easier to review and the git history easier to read

Reply:
Done